pkgs/servers/peertube/default.nix

```diff
@@ -0,0 +1,121 @@
+{ light ? null, pkgs, stdenv, lib, fetchurl, fetchFromGitHub, callPackage
+, nodejs ? pkgs.nodejs-14_x
```

Comment:
```suggestion
, nodejs
```

Reply:
What's wrong with this line? It's currently the default, but I wouldn't expect the team to keep track of which pkgs are expecting which defaults. Or is it bad that we're also passing it in in all-packages?

Reply:
I think you misunderstood what the `?` operator does. `name ? e` provides a default value in the attribute set argument which is used in the case the name is missing.
```nix
nix-repl> f = { n ? "default" }: n
nix-repl> f { n = "non default"; }
"non default"
nix-repl> f { }
"default"
```
That's it. In your case, the attribute set argument is generated by the `callPackage` mechanism, not manually. If `callPackage` knows a package named `nodejs` it will provide it to your function. This package name will always be around so this `?` construct does not what you intended to do.
If you really want 14.x version, just use `nodejs-14_x` in your attribute set :)
https://nixos.org/manual/nix/stable/#ss-functions

Reply:
Then you can remove `light ? null, pkgs`.